primitives/src/valid_proof_of_investor.rs

```diff
@@ -1,38 +1,27 @@
 use crate::{
-    scalar_blake2_from_bytes, CddId, Claim, Context, IdentityId, InvestorZKProofData, Proposition,
-    Ticker,
+    scalar_blake2_from_bytes, CddId, Claim, IdentityId, InvestorZKProofData, Scope, Ticker,
 };
 use cryptography::claim_proofs::ProofPublicKey;
 use curve25519_dalek::{ristretto::CompressedRistretto, scalar::Scalar};
 
 // ZKProofs claims
 // =========================================================
 
-/// Proposition that checks if any of its internal claims exists in context.
+/// Data structure that checks if any of its internal claims exists in context.
 #[derive(Clone)]
 #[cfg_attr(feature = "std", derive(Debug))]
-pub struct ValidProofOfInvestorProposition {
-    /// The Investor proof should be associated to this ticker.
-    pub ticker: Ticker,
-}
-
-impl Proposition for ValidProofOfInvestorProposition {
-    /// Evaluate proposition against `context`.
-    fn evaluate(&self, context: &Context) -> bool {
-        context
-            .claims
-            .iter()
-            .any(|claim| self.evaluate_claim(claim, context))
-    }
-}
+pub struct ValidProofOfInvestor;
 
-impl ValidProofOfInvestorProposition {
+impl ValidProofOfInvestor {
     /// Evaluates if the claim is a valid proof.
-    fn evaluate_claim(&self, claim: &Claim, context: &Context) -> bool {
+    pub fn evaluate_claim(claim: &Claim, id: &IdentityId) -> bool {
         match claim {
-            Claim::InvestorZKProof(ref _ticker_scope, ref scope_id, ref cdd_id, ref proof) => {
-                let message = InvestorZKProofData::make_message(&context.id, &self.ticker);
-                Self::verify_proof(cdd_id, &context.id, scope_id, &self.ticker, proof, &message)
+            Claim::InvestorZKProof(ref scope, ref scope_id, ref cdd_id, ref proof) => {
```

Comment:
`cdd_id` is not verified to be the same as the one authorized by the cdd provider. 

Anyone can use any cdd_id to generate any valid `InvestorZKProof`.

People should be allowed to attach claims/proofs only for their cdd_id.